### DISCUSSION TITLE
How to free exclusive lock of Mariadb

### DISCUSSION THREAD

**Discussion by davidetoile** [photoprism/photoprism#3523] [Q&A] [answered]
Hi,
Since I had to do a cold reboot of my VM, when I run my docker file, I have a mysql loop with this error:
```
[ERROR] mysqld: Can't lock aria control file '/var/lib/mysql/aria_log_control' for exclusive use, error: 11

 | 2023-07-06 16:08:03 0 [Note] Starting MariaDB 10.9.7-MariaDB-1:10.9.7+maria~ubu2204 source revision 33fd519ca7318cd85bee56e8f79df4608ead194e as process 1
photoprism-mariadb-1     | 2023-07-06 16:08:03 0 [ERROR] mysqld: Can't lock aria control file '/var/lib/mysql/aria_log_control' for exclusive use, error: 11. Will retry for 30 seconds
photoprism-photoprism-1  | time="2023-07-06T16:08:11Z" level=debug msg="config: overriding config with values from /photoprism/storage/config/options.yml"
photoprism-photoprism-1  | time="2023-07-06T16:08:11Z" level=debug msg="settings: loaded from /photoprism/storage/config/settings.yml"
photoprism-photoprism-1  | time="2023-07-06T16:08:11Z" level=debug msg="config: running on 'Intel(R) Atom(TM) CPU N2800   @ 1.86GHz', 4.1 GB memory detected"
photoprism-mariadb-1     | 2023-07-06 16:08:33 0 [ERROR] mysqld: Got error 'Could not get an exclusive lock; file is probably in use by another process' when trying to use aria control file '/var/lib/mysql/aria_log_control'
photoprism-mariadb-1     | 2023-07-06 16:08:33 0 [ERROR] Plugin 'Aria' init function returned error.
photoprism-mariadb-1     | 2023-07-06 16:08:33 0 [ERROR] Plugin 'Aria' registration as a STORAGE ENGINE failed.
photoprism-mariadb-1     | 2023-07-06 16:08:33 0 [Note] InnoDB: Compressed tables use zlib 1.2.11
photoprism-mariadb-1     | 2023-07-06 16:08:33 0 [Note] InnoDB: Number of transaction pools: 1
photoprism-mariadb-1     | 2023-07-06 16:08:33 0 [Note] InnoDB: Using generic crc32 instructions
photoprism-mariadb-1     | 2023-07-06 16:08:33 0 [Note] mysqld: O_TMPFILE is not supported on /tmp (disabling future attempts)
photoprism-mariadb-1     | 2023-07-06 16:08:33 0 [Note] InnoDB: Initializing buffer pool, total size = 512.000MiB, chunk size = 8.000MiB
photoprism-mariadb-1     | 2023-07-06 16:08:33 0 [Note] InnoDB: Completed initialization of buffer pool
photoprism-mariadb-1     | 2023-07-06 16:08:33 0 [Note] InnoDB: File system buffers for log disabled (block size=512 bytes)
photoprism-mariadb-1     | 2023-07-06 16:08:33 0 [Note] InnoDB: 128 rollback segments are active.
photoprism-mariadb-1     | 2023-07-06 16:08:33 0 [Note] InnoDB: Setting file './ibtmp1' size to 12.000MiB. Physically writing the file full; Please wait ...
photoprism-mariadb-1     | 2023-07-06 16:08:33 0 [Note] InnoDB: File './ibtmp1' size is now 12.000MiB.
photoprism-mariadb-1     | 2023-07-06 16:08:33 0 [Note] InnoDB: log sequence number 7523208157; transaction id 62050054
photoprism-mariadb-1     | 2023-07-06 16:08:33 0 [Note] Plugin 'FEEDBACK' is disabled.
photoprism-mariadb-1     | 2023-07-06 16:08:33 0 [Note] InnoDB: Loading buffer pool(s) from /var/lib/mysql/ib_buffer_pool
photoprism-mariadb-1     | 2023-07-06 16:08:33 0 [ERROR] Could not open mysql.plugin table: "Unknown storage engine 'Aria'". Some plugins may be not loaded
photoprism-mariadb-1     | 2023-07-06 16:08:33 0 [ERROR] Failed to initialize plugins.
```
Do you know how to fix that? Any clue?
Thanks the community!
Best
David

**Comment by lastzero** [comment]
Sorry for the wait! I did a little research for you and the error seems to be related to the binary log files. If you did a cold restart, that's to be expected.

- To be sure, you should still check if another MariaDB instance is running in your VM.
- If not, stop all services, move the [`aria_log*` and `ib_logfile*` files](https://askubuntu.com/questions/857713/mariadb-wont-start-on-ubuntu-cant-lock-aria-control-file) to a temporary folder and start the MariaDB service.
- If this works and you notice no other issues, it should be safe to delete these files.
- You may also want to disable binary logging in the MariaDB configuration so it doesn't happen again.

**Reply by davidetoile** [reply]
Hello @lastzero 
Thanks for your precious help. I am a new with docker...
I use portainer.io to access to console. As the image reboots all the time, my login in the console using portainer is always rebooted. I do not know how to access to mariadb console of the image to stop the services and move the files. Any hint to do so?
Many thanks in advance
David

**Reply by davidetoile** [reply, marked as answer]
Finally I make it. I had to go to subfolder database in photoprism and move all the indicated files to a temporary folder. Then, system asked for ib_logfile after having recreated all aria_log* files. I had to copy the ib_logfile back to database and restarted all. Then, magic!
Thanks so much, @lastzero !

**Reply by lastzero** [reply]
Excellent, glad you got it working! :)

**Comment by BrandonMFong** [comment]
I've been trying to figure out why my photoprism wasn't working. I run two instances, one for me and another for my partner and I. I saw the error in logs: `[ERROR] mysqld: Can't lock aria control file '/var/lib/mysql/aria_log_control' for exclusive use, error: 11` and this discussion board helped me fix the issue!  Thank you!
